src/rnr_score_hv/innov_netcdf.py

```diff
@@ -0,0 +1,329 @@
+
+                self.metrics_meta.append(metric_meta)
+            except Exception as err:
+                msg = f'Problem creating metrics_meta dataclass - err {err}'
+                raise ValueError(msg) from err
+
+        print(f'metrics_meta: {self.metrics_meta}')
+
+    def set_stats(self):
+        """
+        set the metric meta data specified by the config dict.  the
+        metric meta data helps define the metric file location
+        """
+        try:
+            self.stats = self.config_data.get('stats')
+            for stat in self.stats:
+                if stat not in VALID_STATS:
+                    msg = f'Invalid stat, must be one of \'{VALID_STATS}\''
+                    raise KeyError(msg)
+        except Exception as err:
+            msg = f'\'stats\' key missing, must be one of ' \
+                f'({VALID_STATS}) - err: {err}'
+            raise KeyError(msg) from err
+
+
+    def set_regions(self):
+        """
+        Configure region names and latitude bounds. Use default
+        regions if none are specified.
+        """
+        try:
+            regions = self.config_data.get('regions')
+            if regions is not None:
+
+                for name, region in regions.items():
+
+                    lat_min = float(region.get('lat_min'))
+                    lat_max = float(region.get('lat_max'))
+                    self.regions.append(Region(name, lat_min, lat_max))
+            else:
+                self.regions = DEFAULT_REGIONS
+
+        except Exception as err:
+            msg = f'Problem parsing regions: {regions} - err: {err} '
+            raise ValueError(msg) from err
+
+
+@dataclass
+class InnovStatsConfig(ConfigInterface):
+    """
+    Dataclass to hold and provide configuration information
+    pertaining how the harvester should read netcdf information
+
+    Parameters:
+    -----------
+    config_data: dict - contains configuration data parsed from
+                 either an input yaml file or input dict
+
+    file_meta: FileMeta - parsed configuration data
+    """
+
+    config_data: dict = field(default_factory=dict)
+    harvest_config: dict = field(default_factory=dict, init=False)
+
+    def __post_init__(self):
+        self.harvest_config = self.set_config()
+
+    def set_config(self):
+        ''' initialize HarvestConfig from config_data dict '''
+        return HarvestConfig(self.config_data)
+
+    def get_metrics_meta(self):
+        ''' return all configured MetricsMeta objects '''
+        return self.harvest_config.metrics_meta
+
+    def get_stats(self):
+        ''' return list of all stat types based on harvest_config '''
+        return self.harvest_config.stats
+
+    def get_regions(self):
+        ''' return list of region  '''
+        return self.harvest_config.regions
+
+
+HarvestedData = namedtuple(
+    'HarvestedData',
+    [
+        'cycletime',
+        'region_name',
+        'region_min_lat',
+        'region_max_lat',
+        'plev',
```

Comment:
This is where you might need to generalize. atmosphere is stored on plev but ocean i think is depth. This can be added in a follow up PR that expands to ocean innovation files.

Reply:
we could simply change the column name and use level or elevation instead.  Would that work?

Reply:
yes. like a verticalCoordinate. will have to store units some place else